### ISSUE TITLE
No callback triggered when calling confirmSetupIntent with paypal (have to manually close it)

### ISSUE THREAD

**Issue by singhagam1** [stripe/stripe-react-native#1443]
**Describe the bug**
NOTE - I am very well aware that paypal is in Beta and that Paypal is enabled for the account that I am using.
The code that I am going to mention is working fine in android but in iOS, when the inapp browser opens up for authorisation or deny, nothing happens unless I manually close the inapp browser. After the Inapp browser is manually closed, it then givens the callback. But in android, as soon as I press authorise or deny, the inappbrowser closes automatically and I get the results.

**To Reproduce**
1. Simply call the below function - 

`const {error, setupIntent} = await confirmSetupIntent(
            {stripeClientSecret},
            {
                paymentMethodType: 'PayPal',
                paymentMethodData: {
                    billingDetails: {
                        email: 'someemail@gmail.com',
                    },
                    mandateData: {
                        customerAcceptance: {
                            online: {
                                ipAddress: '1.1.1.1',
                                userAgent: 'my-agent',
                            },
                        },
                    },
                },
            },
            {setupFutureUsage: 'OffSession'},
        );

        if (error) {
            console.log('Setup confirmation error', error.message);
          } else if (setupIntent) {
            console.log('Success from promise', setupIntent);
          }`
2. Open this in ios, an inapp browser will open up as soon as you call this function, after choosing authorise or deny, nothing will happen untill you manually close the inapp browser.

**Expected behavior**
Similar to android, the ios inapp browser should close automatically if I am not missing something.

**Screenshots**
Attaching Screenshot

**Desktop (please complete the following information):**
 - OS: iOS
 - Browser  safari
 - Version 0.28.0 (stripe react native sdk version)

**Smartphone (please complete the following information):**
 - Device: iPhone 
![Simulator Screenshot - iPhone 14 Pro Max - 2023-07-04 at 15 21 41](https://github.com/stripe/stripe-react-native/assets/63900403/ba6b8603-44dc-49e7-8bbf-f5817ad618f5)
14pro max (Simulator)
 - OS: 16.4
 - Browser NA
![Simulator Screenshot - iPhone 14 Pro Max - 2023-07-04 at 15 22 19](https://github.com/stripe/stripe-react-native/assets/63900403/e9674873-2719-439c-ad1e-9b4c4b9fe670)

**Comment by singhagam1** [comment]
Closing because I had to handle it manually as given in the docs 

https://stripe.com/docs/payments/accept-a-payment?platform=react-native&ui=payment-sheet#react-native-set-up-return-url

**Comment by reidak97** [comment]
I am facing the same issue: how did you fix it manually? 

> Closing because I had to handle it manually as given in the docs

**Comment by agam-colaburate** [comment]
<img width="860" alt="Screenshot 2024-01-02 at 12 26 54 AM" src="https://github.com/stripe/stripe-react-native/assets/78752268/c06c657e-eea2-405d-a878-cceb752018c5">

@reidak97 Check the link that I have pasted in my last comment, it has the exact snippet of code.

**Comment by reidak97** [comment]
> @reidak97 Check the link that I have pasted in my last comment, it has the exact snippet of code.

Yes I did, thanks. I am kinda new to App Development, so i could not figure it out yet to handle it manually. The payment would work but the inApp Browser would not close.
